src/cascadia/TerminalSettingsModel/ApplicationState.cpp

```diff
@@ -60,6 +60,31 @@ using namespace ::Microsoft::Terminal::Settings::Model;
 
 namespace winrt::Microsoft::Terminal::Settings::Model::implementation
 {
+    winrt::hstring WindowLayout::ToJson(const Model::WindowLayout& layout)
+    {
+        JsonUtils::ConversionTrait<Model::WindowLayout> trait;
+        auto json = trait.ToJson(layout);
+
+        Json::StreamWriterBuilder wbuilder;
+        const auto content = Json::writeString(wbuilder, json);
+        return hstring{ til::u8u16(content) };
+    }
+
+    Model::WindowLayout WindowLayout::FromJson(const hstring& str)

```

Comment:
I think it should just throw the exception here, because it would be an application issue if our round trip serialization code doesn't work. It was a pain enough to debug this feature as I was adding it since ApplicationState just eats all serialization errors so sometimes I wouldn't notice that it was broken.